src/Avalonia.Base/StyledPropertyBase.cs

```diff
@@ -27,13 +29,10 @@ public abstract class StyledPropertyBase<TValue> : AvaloniaProperty<TValue>, ISt
             Type ownerType,            
             StyledPropertyMetadata<TValue> metadata,
             bool inherits = false,
-            Func<TValue, bool> validate = null,
-            Action<IAvaloniaObject, bool> notifying = null)
+            Func<TValue, bool>? validate = null,
+            Action<IAvaloniaObject, bool>? notifying = null)
                 : base(name, ownerType, metadata, notifying)
         {
-            Contract.Requires<ArgumentNullException>(name != null);
-            Contract.Requires<ArgumentNullException>(ownerType != null);
-
             if (name.Contains("."))
             {
                 throw new ArgumentException("'name' may not contain periods.");
```

Comment:
Is this condition duplicated here and in AvaloniaProperty.cs ?

Reply:
👍 fixed.